File: Algorithms/Python/Count-and-Say.py
class Solution(object):

    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        s = "1"
        for i in range(n - 1):
            temp, pre = '', s[0]
            count = 1
            for x in s[1:]:
                if x == pre:
                    count += 1
                else:
                    temp += (str(count) + pre)
                    count, pre = 1, x
            temp += (str(count) + pre)
            s = temp
        return s

File: Algorithms/Java/Count and Say.java
public class Solution {
    public String countAndSay(int n) {
        String ans = "1";
		for (int i = 1; i < n; i++) {
			String s = ans;
			ans = "";
			int index = 0;
			for (int j = 0; j < s.length(); j++) {
				int count = 0;
				char curr = s.charAt(index);
				while (index < s.length() && s.charAt(index) == curr) {
					count++;
					index++;
				}
				ans = ans + String.valueOf(count) + String.valueOf(curr);
				if (index == s.length())
					break;
			}
		}
		return ans;
    }
}